src/api/uploads/FolderUpload.js

```diff
@@ -48,32 +51,29 @@ class FolderUpload {
      * @param  {Array} Array<UploadFileWithAPIOptions | File> | FileList
      * @returns {void}
      */
-    buildFolderTree(fileList: Array<UploadFileWithAPIOptions | File> | FileList): void {
+    buildFolderTreeFromWebkitRelativePath(fileList: Array<UploadFileWithAPIOptions | File> | FileList): void {
         // FileList does not natively have forEach, hence this workaround
         Array.prototype.forEach.call(fileList, (fileData) => {
-            const file = this.areAPIOptionsInFiles ? fileData.file : fileData;
-            const fileAPIOptions = this.areAPIOptionsInFiles ? fileData.options : {};
+            const file = getFile(fileData);
+            // $FlowFixMe webkitRelativePath is checked below
```

Comment:
Would it be useful to use a different flowtype and append `webkitRelativePath` to the type with intersection &

UploadFile = File & { webkitRelativePath?: string }